pylint/checkers/variables.py

```diff
@@ -2209,9 +2209,8 @@ def _loopvar_name(self, node: astroid.Name) -> None:
         # scope lookup rules would need to be changed to return the initial
         # assignment (which does not exist in code per se) as well as any later
         # modifications.
-        # pylint: disable-next=too-many-boolean-expressions
         if (
-            not astmts
+            not astmts # pylint: disable=too-many-boolean-expressions
```

Comment:
Is this an acceptable change in behaviour? I think it is. This seems to be an unintended side effect of checking for block lines that we didn't consider previously.

Reply:
Is there a way to make disable-next work on the whole if like before ? I think it's going to be reported as a regression, having to change disables when upgrading pylint does not feel great.

Reply:
Not if we also don't want to have the issue reported in the OP. I think it makes sense for `disable-next` only to work for the next line and I think this is a relatively minor change to have to fix in an auto-update. Obviously it is annoying but let's fix this now while the impact is still relatively minor (compared to a 10 year long supported behaviour).

Reply:
Fair enough !